docs/development/setup.md

```diff
@@ -0,0 +1,156 @@
+# Local Development Setup
+
+- [Prerequisites](#prerequisites)
+- [Preperation](#preperation)
+- [Run libvirt-provider for local development](#run-libvirt-provider-for-local-development)
+- [Interact with the `libvirt-provider`](#interact-with-the-libvirt-provider)
+- [Deploy `libvirt-provider`](#deploy-libvirt-provider)
+
+> ℹ️ **NOTE**:</br>
+> To be able to take exec console of the machine, you can follow either of the below approaches:</br>
+> - Run the `libvirt-provider` as the `libvirt-qemu` user.</br>
+> - Create an entry with `devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=0660 0 0` in `/etc/fstab`.</br>
+> - Manually ensure that you have `0660` access permissions on the character files created in `/dev/pts`.</br>
+
+## Prerequisites
+
+- Linux (code contains OS specific code)
+- go >= 1.20
+- `git`, `make` and `kubectl`
+- [Kustomize](https://kustomize.io/)
+- Access to a Kubernetes cluster ([Minikube](https://minikube.sigs.k8s.io/docs/), [kind](https://kind.sigs.k8s.io/) or a
+  real cluster)
+- [libvirt](http://libvirt.org)
+- [QEMU](https://www.qemu.org/download/)
+- `irictl-machine` should be running locally or as [container](https://github.com/ironcore-dev/ironcore/pkgs/container/ironcore-irictl-machine)
+
+## Preperation
+
+### Setup `irictl-machine`
+
+**1. Clone ironcore repository**
+
+```bash
```

Comment:
i think this can be do better

Reply:
Could you please give more context here.

Reply:
i want to show you it offline because it is more changes.

Reply:
Changes are made